src/basic/xattr-util.c

```diff
@@ -133,6 +135,30 @@ int fgetxattrat_fake(
                 xsprintf(fn, "/proc/self/fd/%i", fd);
         }
 
+        /* Pass the FD to the caller, since in case we do openat() the filename depends on it. */
+        *ret_fd = TAKE_FD(fd);
+        strcpy(ret_fn, fn);
```

Comment:
why copy this? why not write the string directly into the buffer via sprintf()?

Reply:
xsprintf doesn't know the size of the buffer so it refuses - I thought we wanted xsprintf especially for perf reasons here?